ClientKit/Display.cs

```diff
@@ -0,0 +1,417 @@
+                "[OSVR] DisplayConfig.GetProjectionForViewerEyeSurface(): native osvrClientGetProjectionForViewerEyeSurface call failed.");
+            return ret;
+        }
+
+        /// <summary>
+        /// Get the projection matrix for a surface seen by an eye of a viewer
+        /// in a display config. (float version)
+        /// </summary>
+        /// <param name="viewer">Viewer ID</param>
+        /// <param name="eye">Eye ID</param>
+        /// <param name="surface">Surface ID</param>
+        /// <param name="near">Distance to near clipping plane - must be nonzero, typically positive.</param>
+        /// <param name="far">Distance to far clipping plane - must be nonzero, typically
+        ///     positive and greater than near.</param>
+        /// <param name="flags">Bitwise OR of matrix convention flags (see OSVR_MatrixFlags)</param>
+        public Matrix44f GetProjectionMatrixForViewerEyeSurfacef(ViewerCount viewer, EyeCount eye, SurfaceCount surface, float near, float far, MatrixConventionsFlags flags)
+        {
+            Matrix44f ret;
+            CheckSuccess(
+                DisplayConfigNative.osvrClientGetViewerEyeSurfaceProjectionMatrixf(mHandle, viewer, eye, surface, near, far, flags, out ret),
+                "[OSVR] DisplayConfig.GetProjectionForViewerEyeSurface(): native osvrClientGetProjectionForViewerEyeSurface call failed.");
+            return ret;
+        }
+
+        /// <summary>
+        /// Determines if a surface seen by an eye of a viewer in a display
+        /// config requests some distortion to be performed.
+        /// 
+        /// This simply reports true or false, and does not specify which kind of
+        /// distortion implementations have been parameterized for this display. For
+        /// each distortion implementation your application supports, you'll want to
+        /// call the function to retrieve the parameters to find out if it is available.
+        /// </summary>
+        /// <param name="viewer">Viewer ID</param>
+        /// <param name="eye">Eye ID</param>
+        /// <param name="surface">Surface ID</param>
+        public bool DoesViewerEyeSurfaceWantDistortion(ViewerCount viewer, EyeCount eye, SurfaceCount surface)
+        {
+            bool ret;
+            CheckSuccess(
+                DisplayConfigNative.osvrClientDoesViewerEyeSurfaceWantDistortion(mHandle, viewer, eye, surface, out ret),
+                "[OSVR] DisplayConfig.DoesViewerEyeSurfaceWantDistortion(): native osvrClientDoesViewerEyeSurfaceWantDistortion call failed.");
+            return ret;
+        }
+
+        /// <summary>
+        /// Returns the radial distortion parameters, if known/requested, for a
+        /// surface seen by an eye of a viewer in a display config.
+        /// 
+        /// NOTE: This method will throw an exception if the surface does not have

```

Comment:
The managed wrapper is returning a value instead of taking an ```out``` argument and returning a status code, as the C API does. I could make it do that, but it's not as nice of an API to use. I would prefer a separate method for each distortion type to check if the HMD supports that distortion type. We can still have the osvrClientDoesViewerEyeSurfaceWantDistortion method, which would return true if any distortion method is wanted by the HMD, and osvrClientDoesViewerEyeSurfaceWantRadialDistortion if the HMD wants radial distortion specifically.

Alternatively, we can indicate the preferred distortion method at the same time if we design the API to return a priority index for the radial distortion method. A priority index value of -1 (or any value less than 0) indicates that radial distortion is not supported and that there are no parameters for it. A value of 0 or greater is that distortion type's relative priority. For example, say an HMD supports two distortion type, A and B. The priority of A is 0, and B is 1, so in this case both A and B are supported, and B is the preferred distortion type (higher numbers are higher priority).